tests/integration/cli/tests/create_exe.rs

```diff
@@ -160,3 +223,187 @@ fn create_exe_works_with_file() -> anyhow::Result<()> {
+
+    let wasm_path = operating_dir.join(create_exe_test_wasm_path());
+    #[cfg(not(windows))]
+    let executable_path = operating_dir.join("wasm.out");
+    #[cfg(windows)]
+    let executable_path = operating_dir.join("wasm.exe");
+
+    let output: Vec<u8> = WasmerCreateExe {
+        current_dir: operating_dir.clone(),
+        wasm_path,
+        native_executable_path: executable_path.clone(),
+        compiler: Compiler::Cranelift,
+        extra_cli_flags: vec!["--object-format", "serialized"],
+        ..Default::default()
+    }
+    .run()
+    .context("Failed to create-exe wasm with Wasmer")?;
+
+    let result = run_code(
+        &operating_dir,
+        &executable_path,
+        &["--eval".to_string(), "function greet(name) { return JSON.stringify('Hello, ' + name); }; print(greet('World'));".to_string()],
+    )
+    .context("Failed to run generated executable")?;
+    let result_lines = result.lines().collect::<Vec<&str>>();
+    assert_eq!(result_lines, vec!["\"Hello, World\""],);
+
+    let output_str = String::from_utf8_lossy(&output);
+    assert!(
+        output_str.contains("erialized"),
+        "create-exe output doesn't mention `serialized` format keyword:\n{}",
+        output_str
+    );
+
+    Ok(())
+}
+
+fn create_obj(args: Vec<&'static str>, keyword_needle: &str, keyword: &str) -> anyhow::Result<()> {
+    let temp_dir = tempfile::tempdir()?;
+    let operating_dir: PathBuf = temp_dir.path().to_owned();
+
+    let wasm_path = operating_dir.join(create_exe_test_wasm_path());
+
+    #[cfg(not(windows))]
+    let object_path = operating_dir.join("wasm.o");
+    #[cfg(windows)]
+    let object_path = operating_dir.join("wasm.obj");
+
+    let output: Vec<u8> = WasmerCreateObj {
+        current_dir: operating_dir.clone(),
+        wasm_path,
+        output_object_path: object_path.clone(),
+        compiler: Compiler::Cranelift,
+        extra_cli_flags: args,
+        ..Default::default()
+    }
+    .run()
+    .context("Failed to create-obj wasm with Wasmer")?;
+
+    assert!(
+        object_path.exists(),
+        "create-obj successfully completed but object output file `{}` missing",
+        object_path.display()
+    );
+    let mut object_header_path = object_path.clone();
+    object_header_path.set_extension("h");
+    assert!(
+        object_header_path.exists(),
+        "create-obj successfully completed but object output header file `{}` missing",
+        object_header_path.display()
+    );
+
+    let output_str = String::from_utf8_lossy(&output);
+    assert!(
+        output_str.contains(keyword_needle),
+        "create-obj output doesn't mention `{}` format keyword:\n{}",
+        keyword,
+        output_str
+    );
+
+    Ok(())
+}
+
+#[test]
+fn create_obj_default() -> anyhow::Result<()> {
+    create_obj(vec![], "ymbols", "symbols")
+}
+
+#[test]
+fn create_obj_symbols() -> anyhow::Result<()> {
+    create_obj(vec!["--object-format", "symbols"], "ymbols", "symbols")
```

Comment:
`ymbols` is this a typo?

Reply:
No, the output contains `Symbols` hence `ymbols` will match uppercase and lowercase if it ever changes

Reply:
Will it match bymbols as well?